src/domain/UBGraphicsTextItem.cpp

```diff
@@ -139,6 +139,44 @@ QFont UBGraphicsTextItem::createDefaultFont()
     return font;
 }
 
+bool UBGraphicsTextItem::event(QEvent *ev)
+{
+    if(ev->type() == QEvent::ShortcutOverride && isSelected())
+    {
+        QKeyEvent* keyEvent = static_cast<QKeyEvent*>(ev);
+        QKeySequence::StandardKey shortcuts[] = {QKeySequence::StandardKey::Bold,
+                                                 QKeySequence::StandardKey::Italic,
+                                                 QKeySequence::StandardKey::Underline};
+        QKeySequence::StandardKey key = QKeySequence::StandardKey::UnknownKey;
+        for(int i=0; i<3; i++)
+            if(keyEvent->matches(shortcuts[i]))
+            {
+                key = shortcuts[i];
+                break;
+            }
+        if(key != QKeySequence::StandardKey::UnknownKey)
+        {
```

Comment:
please simplify this. In `KeyPressEvent`  (didn't test it but something like following) : 
```cpp
// Is checking isSelected really necessary ? Isn't KeyPressEvent (or QGraphicsTextItem::event)
// called only when item is selected or at least on focus ?
if(ev->type() == QEvent::ShortcutOverride && isSelected()) 
{
    if (event->matches(QKeySequence::StandardKey::Bold)
    || event->matches(QKeySequence::StandardKey::Italic)
    || event->matches( QKeySequence::StandardKey::Underline))
    {
        QTextCursor curCursor = textCursor();
        QTextCharFormat format;
        QFont ft(curCursor.charFormat().font());
        if(event->key() == QKeySequence::StandardKey::Bold)
            ft.setBold(!ft.bold());
         else if(event->key() == QKeySequence::StandardKey::Italic)
            ft.setItalic(!ft.italic());
        else
            ft.setUnderline(!ft.underline());
         format.setFont(ft);
         curCursor.mergeCharFormat(format);
         setTextCursor(curCursor);

         contentsChanged();
         event->accept();
         return true; //base class call should be avoided ?
    }
}
```

Reply:
Yes, it is better in the `keyPressEvent()` method. 
But now, for the _Italic_ shortcut, the main 'Ctrl+I' action to set the Pen is throw at first and the  `keyPressEvent()` method never receives the event. So the Italic modification is impossible.
I pushed the modifications to share them because I have no idea to manage this issue. 
Currently, we can:
- set `bold`, `underline` and `italic` in `event()` method
- or only set `bold` and `underline` in `keyPressEvent()` method